src/App.js

```diff
@@ -6,107 +6,61 @@
-  useColorScheme,
-  View,
-} from 'react-native';
+import React, { useState } from 'react';
+import { Button, SafeAreaView, StatusBar, TextInput, View, StyleSheet, Text } from 'react-native';
+import Tictactoe from './Tictactoe';
 
-import {
-  Colors,
-  DebugInstructions,
-  Header,
-  LearnMoreLinks,
-  ReloadInstructions,
-} from 'react-native/Libraries/NewAppScreen';
+const App  = () => {
+  const [text, setText] = useState('');
+  const [size, setSize] = useState(null);
+  const [error, setError] = useState('');
+  const handlePress = () => {
+    if(text < 3 || text > 8 ) {
+      setError('Please enter a number between 3 and 8');
+      return;
+    }
+    setError('');
+    setText('');
+    setSize(+text ? +text : 3);
+  }
 
-const Section = ({children, title}): Node => {
-  const isDarkMode = useColorScheme() === 'dark';
   return (
-    <View style={styles.sectionContainer}>
-      <Text
-        style={[
-          styles.sectionTitle,
-          {
-            color: isDarkMode ? Colors.white : Colors.black,
-          },
-        ]}>
-        {title}
-      </Text>
-      <Text
-        style={[
-          styles.sectionDescription,
-          {
-            color: isDarkMode ? Colors.light : Colors.dark,
-          },
-        ]}>
-        {children}
-      </Text>
-    </View>
-  );
-};
-
-const App: () => Node = () => {
-  const isDarkMode = useColorScheme() === 'dark';
-
-  const backgroundStyle = {
-    backgroundColor: isDarkMode ? Colors.darker : Colors.lighter,
-  };
-
-  return (
-    <SafeAreaView style={backgroundStyle}>
-      <StatusBar barStyle={isDarkMode ? 'light-content' : 'dark-content'} />
-      <ScrollView
-        contentInsetAdjustmentBehavior="automatic"
-        style={backgroundStyle}>
-        <Header />
-        <View
-          style={{
-            backgroundColor: isDarkMode ? Colors.black : Colors.white,
-          }}>
-          <Section title="Step One">
-            Edit <Text style={styles.highlight}>App.js</Text> to change this
-            screen and then come back to see your edits.
-          </Section>
-          <Section title="See Your Changes">
-            <ReloadInstructions />
-          </Section>
-          <Section title="Debug">
-            <DebugInstructions />
-          </Section>
-          <Section title="Learn More">
-            Read the docs to discover what to do next:
-          </Section>
-          <LearnMoreLinks />
-        </View>
-      </ScrollView>
+    <SafeAreaView style={{flex: 1}}>
+      <StatusBar barStyle="dark-content" />
+      {
+        size ? <Tictactoe size={size} setSize={setSize}/> : (
+          <>
+          <View style={styles.container}>
+            <TextInput
+              style={styles.input}
+              onChangeText={setText}
+              placeholder="Enter Game Size"
+            />
+            <Button onPress={handlePress} title="Enter" />
+          </View>
+          {error ? <Text style={styles.error}>{error}</Text> : null}
```

Comment:
also i thought you gonna write it with navigations. better to put in into ur arsenal

Reply:
Ye sure, Once I'll go through the react-navigation, I'll try to implement it here.